resolver/src/main/java/io/netty/resolver/HostsFileNameResolver.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.resolver;
+
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.Promise;
+import io.netty.util.internal.EmptyArrays;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A {@link InetNameResolver} that resolves a host name against the hosts file.
+ * Hosts file is currently only parsed once per instance.
+ */
+public final class HostsFileNameResolver extends InetNameResolver {
+
+    private static UnknownHostException UNMAPPED_HOST_EXCEPTION =
+            new UnknownHostException("Host isn't mapped in hosts file");
+
+    static {
+        UNMAPPED_HOST_EXCEPTION.setStackTrace(EmptyArrays.EMPTY_STACK_TRACE);
+    }
+
+    private final Map<String, InetAddress> hostsFileEntries;
+
+    /**
+     * @param executor the {@link EventExecutor} which is used to notify the listeners of the {@link Future} returned
+     *                 by {@link #resolve(String)}
+     */
+    public HostsFileNameResolver(EventExecutor executor) {
+        this(executor, HostsFileParser.parseSilently());
+    }
+
+    /**
+     * @param executor the {@link EventExecutor} which is used to notify the listeners of the {@link Future} returned
+     *                 by {@link #resolve(String)}
+     * @param hostsFileEntries hostnames that are directly mapped to some {@link InetAddress}
+     */
+    public HostsFileNameResolver(EventExecutor executor, Map<String, InetAddress> hostsFileEntries) {
```

Comment:
hosts files don't support this. See code comment below:

> // trying to map a host to multiple IPs is wrong
// only the first entry is honored